### PR TITLE
[memcached]: Address skipped validation SVR00003

### PR DESCRIPTION
**Pull request by shmsr**
I have written a Go program that is based on `ValiadateKibanaNoDanglingObjectIDs` lint from `package-spec` repository. It uses the same logic to find the dangling references and sets the correct parameters for yq to do in-place changes:

Example:
```sh
$ yq -i -I4 'del(.references.[] | .. | select(.id == "memcached-fleet-managed-default" and .type == "tag"))' kibana/dashboard/memcached-a36fa0b0-eccf-11ec-b66b-6bfdc9ecc703.json
```

I'll soon push that program so that other teams can also use the same to remove any dangling object ids. Very easy to use.

## Proposed commit message

Address SVR00003 for memcached package i.e., removal of dangling object ids.

This is to address the skipped validation part of the package-spec v3 migration.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

Installs successfully and no obvious breakages in viz.

## Related issues

- Relates https://github.com/elastic/integrations/issues/8028

### CODE REVIEW

**Comment by shmsr** [comment]
Should it be a bug-fix or an enhancement? Upon further consideration, I think enhancement here would be more apt?

File: packages/memcached/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.3.1"
+  changes:
+    - description: Address SVR00003 validation skipped during package-spec v3 migration
+      type: bugfix
```
Comment (shmsr) [inline packages/memcached/changelog.yml:5]:
```suggestion
    - description: Remove dangling references from the dashboard
      type: enhancement
```

**Review by aliabbas-elastic — APPROVED** [review]
LGTM

File: packages/memcached/kibana/dashboard/memcached-a36fa0b0-eccf-11ec-b66b-6bfdc9ecc703.json [hunk 2]
```diff
@@ -1069,17 +1069,7 @@
             "id": "metrics-*",
             "name": "5061237a-3a71-485a-be3f-5d8e629e579e:indexpattern-datasource-layer-09bed68a-54a8-41fe-8e2f-6937efc350b4",
             "type": "index-pattern"
-        },
-        {
-            "id": "memcached-fleet-managed-default",
```
Comment (ishleenk17) [inline packages/memcached/kibana/dashboard/memcached-a36fa0b0-eccf-11ec-b66b-6bfdc9ecc703.json:1074]: @shmsr : For my clarity, why do these pointers get created in the first place ?
They become dangling as I suppose the corresponding object gets removed.

Comment (shmsr) [reply packages/memcached/kibana/dashboard/memcached-a36fa0b0-eccf-11ec-b66b-6bfdc9ecc703.json:1074]: If you see the *.json, this is under `references`. References in the dashboard are used to establish relationships between different saved objects or components within Kibana.

Suppose you have 3 different viz that depend on the same index-pattern, so they could reference an index-pattern (e.g., `metrics-*`) that can be used by all 3 viz. So the `id` and `name` will be accordingly and the reference `type` will be `index-pattern`.

So this is a case for tags. There's no relationship currently being maintained with this reference.

If you see: https://github.com/elastic/integrations/pull/8348#issue-1970262963 you'll see some detailed examples of filters and their dependence on references (index-pattern).

**Comment by mrodm** [comment]
Hi @shmsr, please update your branch with the latest contents from main branch. There was an important PR merged updating the CI pipelines. Thanks!

**Comment by shmsr** [comment]
@devamanv Need you review here as well.

**Comment by ishleenk17** [comment]
@shmsr : Still valid ?

**Pull request closed without merging**